packages/gestalt/src/Button.js

```diff
@@ -59,13 +60,17 @@ export default function Button(props: Props): Element<'button'> {
     textColor: textColorProp,
     type = 'button',
   } = props;
+  const { name: colorSchemeName } = useColorScheme();
+  // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
+  const isDarkModeRed = colorSchemeName === 'darkMode' && color === 'red';
 
   const classes = classnames(styles.button, {
     [styles.sm]: size === 'sm',
     [styles.md]: size === 'md',
     [styles.lg]: size === 'lg',
     [styles.solid]: color !== 'transparent',
-    [styles[color]]: !disabled && !selected,
+    [styles[color]]: !disabled && !selected && !isDarkModeRed,
+    [styles.darkModeRed]: isDarkModeRed,
```

Comment:
can we change this to:
```
[styles[isDarkModeRed ? 'darkModeRed' : color]]: !disabled && !selected
```

Reply:
@jackhsu978 Yes, this wasn't quite right thanks for catching! I made a slightly different change that puts the ternary result in a variable instead to hopefully be more readable